com.unity.ml-agents/Runtime/Sensors/VectorSensor.cs

```diff
@@ -8,20 +8,21 @@ namespace Unity.MLAgents.Sensors
     /// <summary>
     /// A sensor implementation for vector observations.
     /// </summary>
-    public class VectorSensor : ISensor, IBuiltInSensor
+    public class VectorSensor : ISensor, ITypedSensor, IBuiltInSensor
     {
         // TODO use float[] instead
         // TODO allow setting float[]
         List<float> m_Observations;
         int[] m_Shape;
         string m_Name;
+        ObservationType m_ObservationType;
 
         /// <summary>
         /// Initializes the sensor.
         /// </summary>
         /// <param name="observationSize">Number of vector observations.</param>
         /// <param name="name">Name of the sensor.</param>
-        public VectorSensor(int observationSize, string name = null)
+        public VectorSensor(int observationSize, string name = null, ObservationType observationType = ObservationType.Default)
```

Comment:
Maybe if `observationType != ObservationType.Default` we should add it to the default name value too?